msg-socket/src/req/driver.rs

```diff
@@ -97,6 +98,18 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Future for ReqDriver<T> {
                 Poll::Pending => {}
             }
 
+            // Check for request timeouts
```

Comment:
This should probably only run on a pre-defined interval, otherwise we'll loop over all pending requests for every request we make. The interval can be something like 1/10 of the configured timeout duration.